qa/tasks/mgr/dashboard/test_pool.py

```diff
@@ -44,8 +44,22 @@ class PoolTest(DashboardTestCase):
     }))
 
     @contextmanager
-    def __create_pool(self, name, data=None):
-        pool_data = data or {
+    def __yield_pool(self, name=None, data=None, deletion_name=None):
+        """
+        Use either just a name or whole description of a pool to create one.
+        This also validates the correct creation and deletion after the pool was used.
+
+        :param name: Name of the pool
+        :param data: Describes the pool in full length
+        :param deletion_name: Only needed if the pool was renamed
+        :return:
+        """
+        data = self._create_pool(name, data)
+        yield data
+        self._delete_pool(deletion_name or data['pool'])
```

Comment:
Nice refactor!

I'd suggest a more meaningful name for the context manager (e.g.: `new_test_pool()` or so) and no need of double underscore here, as `PoolTest` is a final class.

Reply:
I'll rename it :) Thanks for the better name :+1: